### PR TITLE
[DOCS] Add list of supported NLP model architectures

### PR DESCRIPTION
**Pull request by lcawl**
This PR augments the machine learning documentation with a list of NLP model architectures and third party models.

### Preview

* https://stack-docs_1989.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-model-ref.html
* Link in https://stack-docs_1989.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-deploy-models.html#ml-nlp-select-model
* Link in https://stack-docs_1989.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-overview.html

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 1]
```diff
@@ -23,8 +23,29 @@ Per the <<ml-nlp-overview>>, there are multiple ways that you can use NLP
...
+
+The current list of supported architectures is:
+
+* BERT
+* DPR bi-encoders with a Torch wrapper
+* DistilBERT with a Torch wrapper
+* ELECTRA
+* MobileBERT
+* RetriBERT
+* MPNet
```
Comment (joshdevins) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:42]: Is this PR for 8.0? I think this is only in 8.1.

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 2]
```diff
@@ -23,8 +23,29 @@ Per the <<ml-nlp-overview>>, there are multiple ways that you can use NLP
...
+* SentenceTransformers bi-encoders with the above transformer architectures and
+a Torch wrapper
+
+Any trained model that uses one of these supported architectures is deployable
```
Comment (joshdevins) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:46]: I think a clarification about "with a Torch wrapper" means is in order. Basically this means that eland will wrap an existing model automatically in order to make these architectures compatible with the BERT model interface.

**Review by joshdevins — CHANGES_REQUESTED** [review]

**Comment by davidkyle** [comment]
Sorry Lisa, I left this to think about what to do with "with a Torch wrapper".

"with a Torch wrapper" is not common vernacular it won't mean anything to anyone outside of the ML team. 

> I think a clarification about "with a Torch wrapper" means is in order.

This is something that eland does automatically, one of the many reasons we are encouraging users to upload models with eland rather than doing it the hard way with the Elasticsearch APIs. We haven't really explained how to upload a model without using eland but we will have to do that one day, I'm thinking of someone who fine-tunes a model and wants to deploy it to elasticsearch without uploading it to Hugging Face first.



@josh I'd prefer not to clarify what "with a Torch wrapper" means here and save that detail for some in-depth coverage of uploading a model the hard way using the Elasticsearch APIs. Let's keep pushing people down the path of using eland exclusively for now.

**Comment by lcawl** [comment]
> @joshdevins I'd prefer not to clarify what "with a Torch wrapper" means here and save that detail for some in-depth coverage of uploading a model the hard way using the Elasticsearch APIs. Let's keep pushing people down the path of using eland exclusively for now.

Does this mean we should (1) remove the "with a Torch wrapper" clauses from this PR, and (2) add a clarification that this information assumes you're using eland? e.g." Any trained model that uses one of these supported architectures is deployable in Elasticsearch, specifically when you're using eland" or something along those lines?

**Comment by davidkyle** [comment]
> "Any trained model that uses one of these supported architectures is deployable in Elasticsearch, specifically when you're using eland"

Yes, that's my preference if @joshdevins  agrees. We can document the can of worms that is uploading a model without eland later on in a follow up PR.

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc]: I've removed the "with a Torch wrapper" clauses in 749acfcb58f6cc73efdfa752899f218251bcb023

**Comment by lcawl** [comment]
> Yes, that's my preference if @joshdevins agrees.

Thanks, I've removed those clauses from the list and updated the following sentence as well.

**Review by davidkyle — APPROVED** [review]
LGTM

**Review by davidkyle — APPROVED** [review]
LGTM twice

**Pull request merged**